Tests/test_file_apng.py

```diff
@@ -663,6 +665,36 @@ def test_apng_save_blend(tmp_path):
         assert im.getpixel((0, 0)) == (0, 255, 0, 255)
 
 
+def test_save_all_progress():
+    out = BytesIO()
+    progress = []
+
+    def callback(filename, frame_number, n_frames):
+        progress.append((filename, frame_number, n_frames))
+
+    Image.new("RGB", (1, 1)).save(out, "PNG", save_all=True, progress=callback)
+    assert progress == [(None, 1, 1)]
+
+    out = BytesIO()
+    progress = []
+
+    with Image.open("Tests/images/apng/single_frame.png") as im:
+        with Image.open("Tests/images/apng/delay.png") as im2:
+            im.save(
+                out, "PNG", save_all=True, append_images=[im, im2], progress=callback
+            )
+
+    assert progress == [
+        ("Tests/images/apng/single_frame.png", 1, 7),
+        ("Tests/images/apng/single_frame.png", 2, 7),
```

Comment:
This line could use a comment that `single_frame.png` gets appended to itself in the `save()` call

Reply:
> come to think of it, having explicit index there would make adding a comment somewhat redundant

You no longer think this is necessary?